implementations/rust/ockam/ockam_identity/src/channel/decryptor_worker.rs

```diff
@@ -670,31 +627,57 @@ impl Worker for DecryptorWorker {
     ) -> Result<()> {
         let msg_addr = msg.msg_addr();
 
-        match self.state()? {
-            State::KeyExchange => {
-                if msg_addr == self.addresses.decryptor_remote {
-                    self.handle_key_exchange_msg(ctx, msg).await?;
-                } else {
-                    return Err(IdentityError::UnknownChannelMsgDestination.into());
-                }
+        // since once initialized, the state can't change anymore we just borrow a mutable,
+        // doing so we avoid extra copies (in case the compiler don't optimize them away),
+        // and we make sure the state remain initialized even in the case of an error
```

Comment:
The drawaback is that now we don't have a full pattern match on [L.676](https://github.com/build-trust/ockam/pull/4498/files#diff-9faefb775a440129e0a51b66c188674bd94071bed6d24095bdf7439be705d90dR676) :-(.

Reply:
To make it more explicit, I've replaced the returned error with an `unreachable!`